#! /bin/sh /usr/share/dpatch/dpatch-run
## 01-gcc-4.4-fixes.dpatch
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix compilation with GCC 4.4.  (See Debian bug #505589)

@DPATCH@
Index: libsmbios-2.0.3.dfsg/libraries/cmos/CmosRW.cpp
===================================================================
--- libsmbios-2.0.3.dfsg.orig/libraries/cmos/CmosRW.cpp	2010-08-22 16:22:18.334443521 +0200
+++ libsmbios-2.0.3.dfsg/libraries/cmos/CmosRW.cpp	2010-08-22 16:23:04.641547872 +0200
@@ -19,6 +19,7 @@
 // compat header should always be first header if including system headers
 #define LIBSMBIOS_SOURCE
 #include "smbios/compat.h"
+#include <stdio.h>
 #include <errno.h>
 #include <string.h>
 #include "CmosRWImpl.h"
Index: libsmbios-2.0.3.dfsg/libraries/memory/Memory_Linux.cpp
===================================================================
--- libsmbios-2.0.3.dfsg.orig/libraries/memory/Memory_Linux.cpp	2010-08-22 16:22:12.421455887 +0200
+++ libsmbios-2.0.3.dfsg/libraries/memory/Memory_Linux.cpp	2010-08-22 16:22:54.913488246 +0200
@@ -22,6 +22,7 @@
 
 #include <errno.h>
 #include <sys/mman.h>   /* mmap */
+#include <stdio.h>
 #include <unistd.h>     /* getpagesize */
 #include <string.h>     /* memcpy etc. */
 
Index: libsmbios-2.0.3.dfsg/libraries/smbios/SmbiosStrategy_Linux.cpp
===================================================================
--- libsmbios-2.0.3.dfsg.orig/libraries/smbios/SmbiosStrategy_Linux.cpp	2010-08-22 16:22:07.898484146 +0200
+++ libsmbios-2.0.3.dfsg/libraries/smbios/SmbiosStrategy_Linux.cpp	2010-08-22 16:22:41.417947931 +0200
@@ -20,6 +20,7 @@
 #define LIBSMBIOS_SOURCE
 #include "smbios/compat.h"
 
+#include <cstdio>
 #include <sstream>
 #include <string.h>
 
Index: libsmbios-2.0.3.dfsg/libraries/smi/SmiImpl.h
===================================================================
--- libsmbios-2.0.3.dfsg.orig/libraries/smi/SmiImpl.h	2010-08-22 16:21:58.390953094 +0200
+++ libsmbios-2.0.3.dfsg/libraries/smi/SmiImpl.h	2010-08-22 16:22:29.693519469 +0200
@@ -22,6 +22,8 @@
 #include "SmiLowLevel.h"
 #include "ExceptionImpl.h"
 
+#include <cstdio>
+
 namespace smi
 {
     // define our exceptions
